### PR TITLE
Fixes #29855 - Add DSL docs for users

### PR DESCRIPTION
**Pull request by ofedoren**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
 require 'digest/sha1'
 
 class User < ApplicationRecord
+  apipie :class, desc: "A class representing #{model_name.human} object" do
+    sections only: %w[all additional]
+    property :id, Integer, desc: 'Numerical ID of the User'
+    property :firstname, String, desc: 'Returns the user first name'
+    property :lastname, String, desc: 'Returns the user Last name'
```
Comment (tbrisker) [inline app/models/user.rb]:
```suggestion
    property :lastname, String, desc: 'Returns the user last name'
```

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+    property :description, String, desc: 'Returns the user description'
+    property :ssh_keys, array_of: SshKey, desc: 'Returns an array of SshKey object'
```
Comment (tbrisker) [inline app/models/user.rb]:
```suggestion
    property :ssh_keys, array_of: SshKey, desc: 'Returns an array of the user's associated SshKey objects'
```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+    property :ssh_authorized_keys, array_of: SshKey, desc: 'Returns an array of all the ssh authorized_keys'
```
Comment (tbrisker) [inline app/models/user.rb]: This is an array of hashes representing the user's keys with their metadata (key, type and comment), not actual SshKey objects

File: app/models/user.rb [hunk 4]
```diff
@@ -1,6 +1,19 @@
...
+    property :login, String, desc: 'Returns the user login'
+    property :mail, String, desc: 'Returns the user mail'
+    property :last_login_on, String, desc: 'Returns the user Last login time, UTC time'
```
Comment (tbrisker) [inline app/models/user.rb]:
```suggestion
    property :last_login_on, String, desc: 'Returns the user last login time, in UTC time zone'
```

**Review by tbrisker — DISMISSED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (ofedoren) [reply app/models/user.rb]: I think this is an array of strings: https://github.com/theforeman/foreman/blob/c3560417748c744d6f64336b1b98102e6a8bb4c1/app/models/concerns/user_usergroup_common.rb#L2-L4
https://github.com/theforeman/foreman/blob/c3560417748c744d6f64336b1b98102e6a8bb4c1/app/models/ssh_key.rb#L55-L58

File: app/models/user.rb [hunk 5]
```diff
@@ -144,6 +144,19 @@ def as_json(options = {})
...
   end
 
+  apipie :class, desc: "A class representing #{model_name.human} object" do
+    sections only: %w[all additional]
+    property :id, Integer, desc: 'Numerical ID of the User'
+    property :firstname, String, desc: 'Returns the user first name'
+    property :lastname, String, desc: 'Returns the user last name'
+    property :description, String, desc: 'Returns the user description'
+    property :ssh_keys, array_of: SshKey, desc: 'Returns an array of the user\'s associated SshKey objects'
+    property :ssh_authorized_keys, array_of: SshKey, desc: 'Returns an array of strings representing the user\'s keys with their metadata (key, type and comment)'
```
Comment (shiramax) [inline app/models/user.rb]: ssh_authorized_keys it's an array of strings containing the ssh key but not the SshKey object

File: app/models/user.rb [hunk 6]
```diff
@@ -144,6 +144,19 @@ def as_json(options = {})
...
+    property :login, String, desc: 'Returns the user login'
+    property :mail, String, desc: 'Returns the user mail'
+    property :last_login_on, String, desc: 'Returns the user last login time, in UTC time zone'
```
Comment (shiramax) [inline app/models/user.rb]: last_login_on returns an ActiveSupport::TimeWithZone object.

**Review by shiramax — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 5 above)
Comment (ofedoren) [reply app/models/user.rb]: Yeap.. I wrote this in the description as well, but forgot to change the type... :facepalm:

**Comment by ofedoren** [comment]
Thanks, @shiramax! Updated.

**Review by shiramax — APPROVED** [review]

**Comment by shiramax** [comment]
Looks good to me :) @tbrisker do you approve this PR?

**Pull request merged**